From e1728a4606b35e6dca0c393f80d0001e704d8393 Mon Sep 17 00:00:00 2001
From: Wenzong Fan <wenzong.fan@windriver.com>
Date: Thu, 28 Dec 2017 09:13:11 +0000
Subject: [PATCH] wr-policy: allow systemd_logind_t relabel user_tmp_t dirs and
 files

Reference sources: selinux-policy-3.13.1-225.6.fc25.src.rpm
Changes come from: policy-f25-base.patch

Fix avc denial:
  avc: denied { relabelfrom } for pid=254 \
  comm="systemd-logind" name="/" dev="tmpfs" ino=20662 \
  scontext=system_u:system_r:systemd_logind_t \
  tcontext=system_u:object_r:user_tmp_t tclass=dir

Upstream-Status: Pending

Signed-off-by: Wenzong Fan <wenzong.fan@windriver.com>
---
 policy/modules/system/systemd.te    | 2 ++
 policy/modules/system/userdomain.if | 5 +++++
 2 files changed, 7 insertions(+)

diff --git a/policy/modules/system/systemd.te b/policy/modules/system/systemd.te
index 7b6617b..e24c308 100644
--- a/policy/modules/system/systemd.te
+++ b/policy/modules/system/systemd.te
@@ -452,6 +452,8 @@ manage_files_pattern(systemd_logind_t, systemd_logind_inhibit_var_run_t, systemd
 manage_fifo_files_pattern(systemd_logind_t, systemd_logind_inhibit_var_run_t, systemd_logind_inhibit_var_run_t)
 init_pid_filetrans(systemd_logind_t, systemd_logind_inhibit_var_run_t, dir, "inhibit")
 
+userdom_manage_tmp_role(system_r, systemd_logind_t)
+
 allow systemd_logind_t systemd_sessions_var_run_t:dir manage_dir_perms;
 allow systemd_logind_t systemd_sessions_var_run_t:file manage_file_perms;
 allow systemd_logind_t systemd_sessions_var_run_t:fifo_file manage_fifo_file_perms;
diff --git a/policy/modules/system/userdomain.if b/policy/modules/system/userdomain.if
index 8cf8ec6..11370a5 100644
--- a/policy/modules/system/userdomain.if
+++ b/policy/modules/system/userdomain.if
@@ -420,6 +420,11 @@ interface(`userdom_manage_tmp_role',`
 	manage_fifo_files_pattern($2, user_tmp_t, user_tmp_t)
 	files_tmp_filetrans($2, user_tmp_t, { dir file lnk_file sock_file fifo_file })
 	userdom_user_runtime_filetrans_user_tmp($2, { dir file lnk_file sock_file fifo_file })
+	relabel_dirs_pattern($2, user_tmp_t, user_tmp_t)
+	relabel_files_pattern($2, user_tmp_t, user_tmp_t)
+	relabel_lnk_files_pattern($2, user_tmp_t, user_tmp_t)
+	relabel_sock_files_pattern($2, user_tmp_t, user_tmp_t)
+	relabel_fifo_files_pattern($2, user_tmp_t, user_tmp_t)
 ')
 
 #######################################
-- 
2.7.4

